master/buildbot/data/buildrequests.py

```diff
@@ -73,6 +88,15 @@ def get(self, resultSpec, kwargs):
         buildrequest = yield self.master.db.buildrequests.getBuildRequest(kwargs['buildrequestid'])
 
         if buildrequest:
+            filters = resultSpec.popProperties() if hasattr(resultSpec, 'popProperties') else []
+            if filters:
+                try:
+                    props = yield self.master.db.buildsets.getBuildsetProperties(buildrequest['buildsetid'])
+                except (KeyError, TypeError):
```

Comment:
Not sure what those exceptions are eating. I think it is saner to forward the exception to the client rather than sending empty response.

Reply:
Agree. I referred some other code in buildbot which does this exception handling. Removed these exception handling in updated commit.